diff --git a/js/rrssb.js b/js/rrssb.js
index f21685e..8a17c3f 100755
--- a/js/rrssb.js
+++ b/js/rrssb.js
@@ -242,7 +242,7 @@
 	 * Event listners
 	 */
 
-	jQuery('.rrssb-buttons a.popup').on('click', function(e){
+	jQuery(document).on('click', '.rrssb-buttons a.popup', function(e) {
 		var _this = jQuery(this);
 		popupCenter(_this.attr('href'), _this.find('.text').html(), 580, 470);
 		e.preventDefault();
@@ -263,5 +263,7 @@
 		rrssbInit();
 	});
 
+	// Make global
+	window.rrssbInit = rrssbInit;
 
 })(window, jQuery);
