diff --git a/advagg_css_compress/yui/CSSMin.inc b/advagg_css_compress/yui/CSSMin.inc
index eb98750..22b78fa 100644
--- a/advagg_css_compress/yui/CSSMin.inc
+++ b/advagg_css_compress/yui/CSSMin.inc
@@ -213,11 +213,23 @@ class CSSmin
 
         // If current settings are higher respect them.
         foreach ($php_limits as $name => $suggested) {
-            $current = $this->normalize_int(ini_get($name));
-            // memory_limit exception: allow -1 for "no memory limit".
-            if ($current > -1 && ($suggested == -1 || $current < $suggested)) {
-                ini_set($name, $suggested);
-            }
+          $current = $this->normalize_int(ini_get($name));
+
+          if ($current > $suggested) {
+            continue;
+          }
+
+          // memoryLimit exception: allow -1 for "no memory limit".
+          if ($name === 'memory_limit' && $current === -1) {
+            continue;
+          }
