diff --git a/feeds.info b/feeds.info
index 8041cbf..e458f43 100644
--- a/feeds.info
+++ b/feeds.info
@@ -12,22 +12,6 @@ files[] = libraries/ParserCSV.inc
 files[] = libraries/http_request.inc
 files[] = libraries/PuSHSubscriber.inc
 
-; Plugins
-files[] = plugins/FeedsCSVParser.inc
-files[] = plugins/FeedsFetcher.inc
-files[] = plugins/FeedsFileFetcher.inc
-files[] = plugins/FeedsHTTPFetcher.inc
-files[] = plugins/FeedsNodeProcessor.inc
-files[] = plugins/FeedsOPMLParser.inc
-files[] = plugins/FeedsParser.inc
-files[] = plugins/FeedsPlugin.inc
-files[] = plugins/FeedsProcessor.inc
-files[] = plugins/FeedsSimplePieParser.inc
-files[] = plugins/FeedsSitemapParser.inc
-files[] = plugins/FeedsSyndicationParser.inc
-files[] = plugins/FeedsTermProcessor.inc
-files[] = plugins/FeedsUserProcessor.inc
-
 ; Tests
 files[] = tests/feeds.test
 files[] = tests/feeds_date_time.test
