diff --git a/jquery.flexslider.js b/jquery.flexslider.js
index eb54c64..01ed11c 100644
--- a/jquery.flexslider.js
+++ b/jquery.flexslider.js
@@ -157,7 +157,7 @@
           slider.currentItem = slider.currentSlide;
           slider.slides.removeClass(namespace + "active-slide").eq(slider.currentItem).addClass(namespace + "active-slide");
           if(!msGesture){
-              slider.slides.on(eventType, function(e){
+              slider.slides.bind(eventType, function(e){
                 e.preventDefault();
                 var $slide = $(this),
                     target = $slide.index();
