diff --git a/src/salvattore.js b/src/salvattore.js
index 2f7607e..7eaf9d5 100644
--- a/src/salvattore.js
+++ b/src/salvattore.js
@@ -337,12 +337,11 @@ self.init = function init() {
 };
 
 
-self.init();
-
 return {
   append_elements: self.append_elements,
   prepend_elements: self.prepend_elements,
-  register_grid: self.register_grid
+  register_grid: self.register_grid,
+  init: self.init
 };
 
 })(window, window.document);
