diff --git a/dist/salvattore.js b/dist/salvattore.js
index d547557..a97a724 100644
--- a/dist/salvattore.js
+++ b/dist/salvattore.js
@@ -612,12 +612,12 @@ self.init = function init() {
 };
 
 
-self.init();
 
 return {
   append_elements: self.append_elements,
   prepend_elements: self.prepend_elements,
-  register_grid: self.register_grid
+  register_grid: self.register_grid,
+  init: self.init
 };
 
 })(window, window.document);
